### PR TITLE
WIP: update README

### PR DESCRIPTION
**Pull request by Remi-Gau**
OK so this is just to get the ball rolling. 

Wanna add more but I want to make sure my fevered brain is not leading off track too much.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,43 @@
-# COBIDAS_chckls_SPM
-Turns a SPM pipeline into a filled-in COBIDAS checklist
+# COBIDAS checklist
+
+This repository is a first step in trying to turn the report published by the Committee on Best Practices in Data Analysis and Sharing (COBIDAS) of OHBM into an easy to use checklist to improve methods and results reporting in neuroimaging.
+
+This should a place where:
```
Comment (cassgvp) [inline README.md]:
```suggestion
This repository should be a place where:
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,43 @@
...
+1. issues and ideas around reaching that goal are discussed,
```
Comment (cassgvp) [inline README.md]:
```suggestion
1. issues and ideas around reaching this goal are discussed;
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,43 @@
...
+2. plans on how to get to this goal are designed and implemented.
+
+## How to reach us
+[The github repository of this project can be found here](https://github.com/Remi-Gau/COBIDAS_chckls)
+
+Come and join us at on the `cobidas_checklist` channel on the [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/).
```
Comment (cassgvp) [inline README.md]:
```suggestion
Come and join us on the `cobidas_checklist` channel at [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/).
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,43 @@
...
+
+A few years ago, in order to improve replicability in f/MRI research, the Committee on Best Practices in Data Analysis and Sharing ([COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728)) of OHBM released a [report](https://www.biorxiv.org/content/10.1101/054262v2) to promote best practices for methods and results reporting. This was recently followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
+
+Contrary to what the most optimistic people might have thought, these guidelines do not seem to have been widely adopted and anecdotal evidence ([see that twitter poll and thread](https://treeverse.app/view/Xf3jfvIZ)) suggests that, even among people who know about the report, few of them use it to write or review papers.
+
+One possible reason for this might have the unwieldy nature of the report. Anticipating on this issue, the authors of the report had included a checklist that still ended up taking almost 30 of the 70 pages of the whole document. Anyone who has used this checklist tends to agree that it is a great resource but that it is a bit cumbersome to use.
```
Comment (cassgvp) [inline README.md]:
```suggestion
One possible reason for this might be the unwieldy nature of the report. Anticipating this issue, the authors of the guidelines included a checklist (Appendix D) that still ended up taking almost 30 of the 70 pages of the whole document. Anyone who has used this checklist tends to agree that it is a great resource but that it is a bit cumbersome to interpret and apply.
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,43 @@
...
+
+### Short term goal
+
+**The short term goal of this project is to make this checklist easier to use**.
+
+One way would be to turn it into a website that people can simply click their way through.
```
Comment (cassgvp) [inline README.md]:
```suggestion
One way to achieve this may be to turn the checklist into a website that users can click through and provide the information requested by COBIDAS.
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,43 @@
...
+
+Ideally this would generate a small text file (for example a json file) that summarize what option was chosen for each item of the checklist.
```
Comment (cassgvp) [inline README.md]:
```suggestion
Ideally this would generate a small text file (for example a json file) that summarizes the response provided for each item of the checklist.
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,43 @@
...
+
+#### Extensions
+
+Filling in that checklist could still be long and tedious so a way to facilitate this further would be read as much metadata as possible from existing standards like:
+1. the [Brain imaging data structure](http://bids.neuroimaging.io/)  used for that study (for example with respect to the data acquisition and experimental design).
+2. the [NIDM results](http://nidm.nidash.org/specs/nidm-results_130.html) of any mass-univariate analysis performed for this study.
+
+#### Further developments
+
+Other off-shoot of this project could include developing toolboxes / packages / plug-ins for the main neuroimaging software (SPM, FSL, AFNI, Freesurfer) that would take some processing batch file as input (e.g matlabbatch.mat for SPM) and would output json files that could be used on the COBIDAS checklist website to automatically generate methods section (in a similar way to [what fMRIprep does](https://fmriprep.readthedocs.io/en/stable/citing.html))
```
Comment (cassgvp) [inline README.md]:
```suggestion
Other off-shoot of this project could include developing toolboxes / packages / plug-ins for the main neuroimaging software (SPM, FSL, AFNI, Freesurfer) that would take a processing batch file as input (e.g matlabbatch.mat for SPM or design.fsf for FSL) and output json files that could be used on the proposed COBIDAS checklist website to automatically generate methods section (in a similar way to [fMRIprep](https://fmriprep.readthedocs.io/en/stable/citing.html))
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,43 @@
-# COBIDAS_chckls_SPM
-Turns a SPM pipeline into a filled-in COBIDAS checklist
+# COBIDAS checklist
+
+This repository is a first step in trying to turn the report published by the Committee on Best Practices in Data Analysis and Sharing (COBIDAS) of OHBM into an easy to use checklist to improve methods and results reporting in neuroimaging.
```
Comment (cassgvp) [inline README.md]:
```suggestion
This repository is a first step in trying to turn the report published by the Committee on Best Practices in Data Analysis and Sharing (COBIDAS) of OHBM into a practical tool for improving methods and results reporting in neuroimaging.
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,43 @@
...
+
+This should a place where:
+1. issues and ideas around reaching that goal are discussed,
+2. plans on how to get to this goal are designed and implemented.
```
Comment (cassgvp) [inline README.md]:
```suggestion
2. plans for how to reach this goal may be designed and implemented.
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,43 @@
...
+
+## How to reach us
+[The github repository of this project can be found here](https://github.com/Remi-Gau/COBIDAS_chckls)
+
+Come and join us at on the `cobidas_checklist` channel on the [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/).
+
+Otherwise you can open a new issue if there is something you would like to discuss directly here.
```
Comment (cassgvp) [inline README.md]:
```suggestion
Otherwise you can open a new issue on the repository itself if there is something you would like to discuss directly here.
```

File: README.md [hunk 11]
```diff
@@ -1,2 +1,43 @@
...
+
+## Project Description
+In 2012, in a [review of the methods and results reporting of more than 200 fMRI paper](https://www.ncbi.nlm.nih.gov/pubmed/22796459) the author found that "_Although many journals urge authors to describe their methods to a level of detail such that independent investigators can fully reproduce their efforts, the results described here suggest that few studies meet this criterion._"
+
+A few years ago, in order to improve replicability in f/MRI research, the Committee on Best Practices in Data Analysis and Sharing ([COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728)) of OHBM released a [report](https://www.biorxiv.org/content/10.1101/054262v2) to promote best practices for methods and results reporting. This was recently followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
```
Comment (cassgvp) [inline README.md]:
```suggestion
In order to address this issue of replicability in f/MRI research, [COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728) released a [guidelines](https://www.biorxiv.org/content/10.1101/054262v2) for best practices in reporting of methods and results. This was soon followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
```

File: README.md [hunk 12]
```diff
@@ -1,2 +1,43 @@
...
+
+Contrary to what the most optimistic people might have thought, these guidelines do not seem to have been widely adopted and anecdotal evidence ([see that twitter poll and thread](https://treeverse.app/view/Xf3jfvIZ)) suggests that, even among people who know about the report, few of them use it to write or review papers.
+
+One possible reason for this might have the unwieldy nature of the report. Anticipating on this issue, the authors of the report had included a checklist that still ended up taking almost 30 of the 70 pages of the whole document. Anyone who has used this checklist tends to agree that it is a great resource but that it is a bit cumbersome to use.
+
+### Short term goal
+
+**The short term goal of this project is to make this checklist easier to use**.
```
Comment (cassgvp) [inline README.md]:
```suggestion
**The short term goal of this project is to make the Appendix D checklist easier to use**.
```

File: README.md [hunk 13]
```diff
@@ -1,2 +1,43 @@
...
+Ideally this would generate a small text file (for example a json file) that summarize what option was chosen for each item of the checklist.
+
+This machine readable file could then be used to automatically generate part of the methods section of an article.
+
+#### Extensions
+
+Filling in that checklist could still be long and tedious so a way to facilitate this further would be read as much metadata as possible from existing standards like:
+1. the [Brain imaging data structure](http://bids.neuroimaging.io/)  used for that study (for example with respect to the data acquisition and experimental design).
+2. the [NIDM results](http://nidm.nidash.org/specs/nidm-results_130.html) of any mass-univariate analysis performed for this study.
+
```
Comment (cassgvp) [inline README.md:60]: Not sure I understand this bit. Are you suggesting reading in the BIDS participants or sequence json files for TR, TE etc.? Perhaps you could give some more explicit examples? I'm also not familiar with NIDM, so examples might be useful here too.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,43 @@
...
+One way would be to turn it into a website that people can simply click their way through.
+
+Ideally this would generate a small text file (for example a json file) that summarize what option was chosen for each item of the checklist.
+
+This machine readable file could then be used to automatically generate part of the methods section of an article.
+
+#### Extensions
+
+Filling in that checklist could still be long and tedious so a way to facilitate this further would be read as much metadata as possible from existing standards like:
```
Comment (cassgvp) [inline README.md]:
```suggestion
The process of filling in an online checklist could be simplified if available metadata was drawn in from machine readable files created in the fulfilment of existing standards, for example:
```

File: README.md (hunk 7 above)
Comment (cassgvp) [inline README.md]: Do you want to include fmriprep as a "main neuroimaging software"? Feels like it would be a relevant application (I know I'm going to be scoring the fmriprep documentation to find details for COBIDAS!)

**Review by cassgvp — COMMENTED** [review]
This looks great, and I really like the breakdown of first application to extensions. Just tweeked the wording a little in a few places, and one suggestion for clarification on the first extension. 

Exciting!

**Comment by cassgvp** [comment]
Hi Remi,

I *think* I've just submitted my review of the readme, but I've not used
github in this way before, so let me know if there is anything I need to do
to finalise it.

Best,

Cass

File: README.md (hunk 7 above)
Comment (Remi-Gau) [reply README.md]: Yeah I am not sure that fMRIprep would fall in that category because it bundles together so many other softwares.
But it would make sense to have an option "data preprocessed with fMRIprep" (and eventually point to a json file because fMRIprep still has options) and it would automatically fill in the corresponding part of the checklist.

Actually that makes me think that having the same possibility for other pre-processing pipelines that exist out there might be a good idea. 

Note to self: think about possibilities to link this more generally with nipype, and the PORCUPINE and GIRAFFE pipeline designing tools that Tim van Mourik has been working on.

Comment (Remi-Gau) [reply README.md]: I will rephrase and extend that.

File: README.md (hunk 13 above)
Comment (Remi-Gau) [reply README.md:60]: Examples are very good idea. Will add some.

File: OHBM_hackathon_2019_project_description.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
+# Improving the COBIDAS cheklist for better neuroimaging methods and results reporting
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]:
```suggestion
# Improving the COBIDAS checklist for better neuroimaging methods and results reporting
```

File: OHBM_hackathon_2019_project_description.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+Remi Gau ([ORCID](https://orcid.org/0000-0002-1535-9767
+))
+
+## Project Description
+In 2012, in his review of the methods and results reporting of more than 200 fMRI paper, [Joshua Carp wrote](https://www.ncbi.nlm.nih.gov/pubmed/22796459): "_Although many journals urge authors to describe their methods to a level of detail such that independent investigators can fully reproduce their efforts, the results described here suggest that few studies meet this criterion._"
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]:
```suggestion
In 2012, in his review of the methods and results reporting of more than 200 fMRI papers, [Joshua Carp wrote](https://www.ncbi.nlm.nih.gov/pubmed/22796459): "_Although many journals urge authors to describe their methods to a level of detail such that independent investigators can fully reproduce their efforts, the results described here suggest that few studies meet this criterion._"
```

File: OHBM_hackathon_2019_project_description.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+A few years ago, in order to improve the situation with  respect to [reproducibility](https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#reproducible-neuroimaging-tools) in f/MRI research, the Committee on Best Practices in Data Analysis and Sharing ([COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728)) of OHBM released a [report](https://www.biorxiv.org/content/10.1101/054262v2) to promote best practices for methods and results reporting. This was recently followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]:
```suggestion
A few years ago, in order to improve the situation with respect to [reproducibility](https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#reproducible-neuroimaging-tools) in f/MRI research, the Committee on Best Practices in Data Analysis and Sharing ([COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728)) of OHBM released a [report](https://www.biorxiv.org/content/10.1101/054262v2) to promote best practices for methods and results reporting. This was recently followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
```

File: OHBM_hackathon_2019_project_description.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+### Short term goals
+
+**The short term goal of this project is to make the Appendix D checklist easier to use**.
+
+One way to achieve this may be to turn the checklist into a website that users can click through and provide the information requested by COBIDAS. Ideally this would generate a small text file (for example a json file) that summarizes what option was chosen for each item of the checklist. This machine readable file could then be used to automatically generate part of the methods section of an article.
+
+A side branch of this project includes developing toolboxes  for the main neuroimaging software (SPM, FSL...) that given a processing batch file as input (e.g matlabbatch.mat for SPM or design.fsf for FSL) creates an output json file readable by the COBIDAS checklist website.
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]:
```suggestion
A side branch of this project includes developing plugins for the main neuroimaging software toolboxes (SPM, FSL, AFNI). Such a plugin would receive processing batch files as input (e.g matlabbatch.mat for SPM or design.fsf for FSL) and would create output files (e.g. in json format) readable by the COBIDAS checklist website, which in turn would be able to display the neuroimage processing pipeline and/or generate part of the methods section for an article.
```

File: OHBM_hackathon_2019_project_description.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+
+Other potential goals (interaction with BIDS and NIDM, further integration with main neuroimaging softwares) and potential applications (improving data-sharing and peer-review) of this project are described in this [repository](https://github.com/Remi-Gau/COBIDAS_chckls).
+
+#### Requirements
+
+The implementation of this project should remain flexible enough to:
+- accommodate the inclusion of new items in the checklist as new neuroimaging methods mature (e.g new multivariate analysis, high-resolution MRI...),
+- easily fork the project and convert it to create a checklist-website for a different field. In practice, this will most likely involve mean a deployment through a container technology like [docker](https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#containers).
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]:
```suggestion
- easily fork the project and convert it to create a checklist-website for a different field. In practice, this will most likely involve a deployment through a container technology like [docker](https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#containers).
```

File: OHBM_hackathon_2019_project_description.md [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Milestones
+
+- Discuss conceptual and structural details of the COBIDAS-json file.
+
+- Create a template of the COBIDAS-json file
+
+- Create a proof of concept website that
+  - generates a checklist to by users from an empty template COBIDAS-json file,
```
Comment (jsheunis) [inline OHBM_hackathon_2019_project_description.md]: This sentence doesn't make sense, but I'm not sure what to change in order to improve it. @Remi-Gau maybe rephrase to highlight the intended meaning?

**Review by jsheunis — COMMENTED** [review]
I mainly fixed typos and suggested word/phrase changes. One important thing that we should consider IMO is to have a single, clear, achievable goal specifically for the hackathon. If this goal is to create the website with a clickable checklist that generates json file at the end, then we might make this explicit in the project description. If it already was clear, then I missed it. There are already a number of really cool ideas that we could include or focus on (e.g. the plugins for SPM/FSL, the generation of methods sections, the creation of a software-specific batch file based on the json cobidas checklist, and many more), but I suspect that this will detract from a central focus needed at hackathon. That is, unless we get A LOT of interest and we have enough people to divide into groups with different focus items. What I also think will be a good idea to do before the hackathon is to figure out what a good software stack or infrastructure would be for the website, plugins, etc, and to make some figure/diagram mockups of how the different components would fit together. This could help us start the hackathon with a plan in place and ready to code. (P.s. as with everything I say, these things are totally up for discussion as I'm never fully convinced by my own arguments.)

**Comment by Remi-Gau** [comment]
I think you are right. I should stick to one goal even if a bit too vague-ish and point to other possibilities by referring back to the main project repo.
